### PR TITLE
Feature/issue37 change title in issue list

### PR DESCRIPTION
**Pull request by yoshio15**
## 💪 Summary
working on #37

## 🏁 Goals
Use `ghr` to change issue title in issue list.

### CODE REVIEW

File: autoload/gh/issues.vim [hunk 1]
```diff
@@ -108,12 +109,30 @@ function! s:set_issue_list(resp) abort
...
+  nmap <buffer> <silent> ghr   <Plug>(gh_issue_edit_title)
   nmap <buffer> <silent> ghy   <Plug>(gh_issue_url_yank)
   call gh#help#set_keymap('issues')
 
   call gh#provider#preview#open(function('s:get_preview_info'))
 endfunction
 
+function! s:edit_issue_title() abort
+  let title_before = gh#provider#list#current().title
+  let userInput = input("Enter title of issue.\nissue title: ", title_before)
```
Comment (skanehira) [inline autoload/gh/issues.vim]: I think variable name `new_title` can be more better

Comment (skanehira) [reply autoload/gh/issues.vim]: I think prompt message `new issue title: xxxxxxxx` be more simple.

File: autoload/gh/issues.vim [hunk 2]
```diff
@@ -108,12 +109,30 @@ function! s:set_issue_list(resp) abort
...
   nmap <buffer> <silent> ghn   <Plug>(gh_issue_new)
+  nmap <buffer> <silent> ghr   <Plug>(gh_issue_edit_title)
   nmap <buffer> <silent> ghy   <Plug>(gh_issue_url_yank)
   call gh#help#set_keymap('issues')
 
   call gh#provider#preview#open(function('s:get_preview_info'))
 endfunction
 
+function! s:edit_issue_title() abort
+  let title_before = gh#provider#list#current().title
```
Comment (skanehira) [inline autoload/gh/issues.vim]: title_before -> current_title

File: autoload/gh/issues.vim [hunk 3]
```diff
@@ -108,12 +109,30 @@ function! s:set_issue_list(resp) abort
...
+  let userInput = input("Enter title of issue.\nissue title: ", title_before)
+
+  if trim(userInput) ==# title_before
+    call gh#gh#error_message('title has not changed')
+    return
+  endif
+
+  let data = { 'title': userInput }
+  let number = gh#provider#list#current().number[1:]
+  call gh#gh#message('issue updating...')
```
Comment (skanehira) [inline autoload/gh/issues.vim]: `issue updating...` -> `issue title updating...`

File: autoload/gh/issues.vim [hunk 4]
```diff
@@ -108,12 +109,30 @@ function! s:set_issue_list(resp) abort
...
 
   call gh#provider#preview#open(function('s:get_preview_info'))
 endfunction
 
+function! s:edit_issue_title() abort
+  let title_before = gh#provider#list#current().title
+  let userInput = input("Enter title of issue.\nissue title: ", title_before)
+
+  if trim(userInput) ==# title_before
+    call gh#gh#error_message('title has not changed')
```
Comment (skanehira) [inline autoload/gh/issues.vim]: We doesn't need to print error message, just do nothing is better I think

**Review by skanehira — CHANGES_REQUESTED** [review]

**Comment by skanehira** [comment]
Thanks :) 
I leaved some comments

**Comment by yoshio15** [comment]
@skanehira 
thank you for some comments.
updated PR :)

**Review by skanehira — APPROVED** [review]
LGTM
Thanks 😄

**Pull request merged**